flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java

```diff
@@ -498,6 +498,20 @@ public class KubernetesConfigOptions {
                             "Whether to enable HostNetwork mode. "
                                     + "The HostNetwork allows the pod could use the node network namespace instead of the individual pod network namespace. Please note that the JobManager service account should have the permission to update Kubernetes service.");
 
+    public static final ConfigOption<Map<String, String>> KUBERNETES_USER_JAR_ARTIFACT_HTTP_HEADER =
+            ConfigOptions.key("kubernetes.user.artifacts.http.header")
+                    .mapType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Custom HTTP header for HttpArtifactFetcher. The header will be applied when getting the application job artifacts. "
+                                    + "Expected format: headerKey1:headerValue1,headerKey2:headerValue2.");
+
+    public static final ConfigOption<String> KUBERNETES_USER_ARTIFACTS_BASE_DIR =
+            ConfigOptions.key("kubernetes.user.artifacts.base.dir")
+                    .stringType()
+                    .defaultValue("/opt/flink/artifacts")
```

Comment:
I have a sense that we could also mount an empty dir for this path. By this, if the container restart, the jar resource will kept. It can be optimized later.

Reply:
Good Idea. It is a good way to keep the jar resource  without declaring pv and pvc .But  ,I think We need to  add a configuration to control whether `empty dir` is enabled.As far as I know, `empty dir` is forbidden in some user k8s cluster.I will create a ticket and attach a PR for this feature after finishing PR `FLINK-28915`.